File: packages/mp_android_runtime/mp_android_runtime/src/main/java/com/mpflutter/runtime/components/basic/IgnorePointer.java
package com.mpflutter.runtime.components.basic;

import android.content.Context;
import android.view.MotionEvent;

import androidx.annotation.NonNull;

import com.mpflutter.runtime.components.MPComponentView;
import com.mpflutter.runtime.jsproxy.JSProxyObject;

import org.json.JSONObject;

public class IgnorePointer extends MPComponentView {

    boolean ignoring = false;

    public IgnorePointer(@NonNull Context context) {
        super(context);
    }

    @Override
    public void setAttributes(JSProxyObject attributes) {
        super.setAttributes(attributes);
        this.ignoring = attributes.optBoolean("ignoring", false);
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent ev) {
        if (ignoring) {
            return true;
        }
        return super.dispatchTouchEvent(ev);
    }
}

File: packages/mp_dom_runtime/src/components/basic/ignore_pointer.ts
import { ComponentView } from "../component_view";
import { setDOMStyle } from "../dom_utils";

export class IgnorePointer extends ComponentView {
  setAttributes(attributes: any) {
    super.setAttributes(attributes);
    setDOMStyle(this.htmlElement, { pointerEvents: attributes.ignoring ? "none" : "unset" });
  }
}
